### PR TITLE
Unified input validation for reductions

### PR DESCRIPTION
**Pull request by romanlutz**

### CODE REVIEW

File: fairlearn/_input_validation.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+_MESSAGE_X_NONE = "Must supply X"
+_MESSAGE_Y_NONE = "Must supply y"
+_MESSAGE_X_Y_ROWS = "X and y must have same number of rows"
+_MESSAGE_X_SENSITIVE_ROWS = "X and the sensitive features must have same number of rows"
+
+_KW_SENSITIVE_FEATURES = "sensitive_features"
+
+
+def _validate_and_reformat_reductions_input(X, y, enforce_binary_sensitive_feature=False,
```
Comment (riedgar-ms) [inline fairlearn/_input_validation.py]: A minor thing, but if the filename has an underscore, you don't need an underscore on the methods therein :-)

Comment (romanlutz) [reply fairlearn/_input_validation.py]: This one is so that people don't start relying on it. The filename was for doc, right?

Comment (riedgar-ms) [reply fairlearn/_input_validation.py]: Well, I don't think people should be looking in an underscored file. It's not a big thing, and does no harm to leave it. I didn't go through everything in reductions and metrics to make sure that files with underscored names only contained underscored things.

Comment (romanlutz) [reply fairlearn/_input_validation.py]: I don't think that's necessary either. I just worry about methods moving between files, so if these suddenly go into a different file then they're not "private" anymore. I don't see any harm in leaving it the way it is right now. If you feel very strongly that it should change I can adjust.

Comment (riedgar-ms) [reply fairlearn/_input_validation.py]: My feelings are definitely not strong on the matter :-)

**Pull request merged**